File: Q7 Chocolate distribution problem.py
class Solution:

    def findMinDiff(self, A, N, M):
        # Sort the array
        A.sort()

        # Initialize the minimum difference
        min_diff = float('inf')

        # Traverse the sorted list and find the minimum difference between the maximum and minimum of M elements
        for i in range(N - M + 1):
            diff = A[i + M - 1] - A[i]
            if diff < min_diff:
                min_diff = diff

   

File: Q7 Chocolate distribution problem.js

class Solution{
    findMinDiff(arr,n,m){
        //code here
        arr.sort((a,b)=>a-b);
        let minDiff = Infinity;

        // Traverse the sorted list and find the minimum difference between the maximum and minimum of m elements
        for (let i = 0; i + m - 1 < n; i++) {
            let diff = arr[i + m - 1] - arr[i];
            if (diff < minDiff) {
                minDiff = diff;
            }
        }

        return minDiff;
    }
}
